upf/upf.c

```diff
@@ -380,6 +380,23 @@ vnet_upf_node_id_set (const pfcp_node_id_t * node_id)
   return VNET_API_ERROR_INVALID_ARGUMENT;
 }
 
+int
+vnet_upf_pfcp_heartbeat_config (u32 timeout, u32 retries)
+{
+  pfcp_server_main_t *psm = &pfcp_server_main;
+
+  if ((timeout > 60) || (retries > 10))
+    return 1;
+
+  if (timeout)
+    psm->hb_cfg.timeout = timeout;
+
+  if (retries)
+    psm->hb_cfg.retries = retries;
```

Comment:
What if I want to set `0` retries for some kind of test?
Wouldn't `~0` make more sense as the "don't update" value?

Reply:
yep, agree. Fixed